cosi/threshold/mod.go

```diff
@@ -47,15 +72,23 @@ func (c *CoSi) GetPublicKeyFactory() crypto.PublicKeyFactory {
 // GetSignatureFactory implements cosi.CollectiveSigning. It returns the
 // signature factory.
 func (c *CoSi) GetSignatureFactory() crypto.SignatureFactory {
-	return SignatureFactory{
-		aggFactory: c.signer.GetSignatureFactory(),
-	}
+	return types.NewSignatureFactory(c.signer.GetSignatureFactory())
 }
 
 // GetVerifierFactory implements cosi.CollectiveSigning. It returns the verifier
 // factory.
 func (c *CoSi) GetVerifierFactory() crypto.VerifierFactory {
-	return verifierFactory{factory: c.signer.GetVerifierFactory()}
+	return types.NewThresholdVerifierFactory(c.signer.GetVerifierFactory())
+}
+
+// SetThreshold implements cosi.CollectiveSigning. It sets a new threshold
+// function.
+func (c *CoSi) SetThreshold(fn cosi.Threshold) {
+	if fn == nil {
```

Comment:
why would fn be nil ?

Reply:
No specific reason but that's a defensive programming approach as it will panic if the type is different.